drivers/javascript/net.coffee

```diff
@@ -781,10 +781,10 @@ class Connection extends events.EventEmitter
             query.global_optargs[util.fromCamelCase(key)] = r.expr(value).build()
 
         # If the user has specified the `db` on the connection (either
-        # in the constructor, or with the `.use` method, we add that
-        # db to the optargs for the query.
-        if @db?
-            query.global_optargs['db'] = r.db(@db).build()
+        # in the constructor, as an optarg to .run(), or with the `.use` method)
+        # we add that db to the optargs for the query.
+        if opts.db? or @db?
+            query.global_optargs.db = r.db(opts.db or @db).build()
```

Comment:
You can slightly simplify this, since the `db` optarg is already added to query.global_optargs above in 781:
```js
if @db? and not opts.db?
  query.global_optargs['db'] = r.db(@db).build()
```

I believe `db` also works if it's passed to the server as a string instead of an r.db object and this is what we are eventually going to move to entirely (compare https://github.com/rethinkdb/docs/issues/332, though it's unclear when that will happen).